internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -666,6 +678,7 @@ func flattenKustoClusterVNET(vnet *clusters.VirtualNetworkConfiguration) []inter
 		"subnet_id":                    vnet.SubnetId,
 		"engine_public_ip_id":          vnet.EnginePublicIPId,
 		"data_management_public_ip_id": vnet.DataManagementPublicIPId,
+		"enabled":                      vnet.State == nil || *vnet.State == clusters.VnetStateEnabled,
 	}
```

Comment:
Would something like this be more clear?

```suggestion
	enabled := true
	if vnet.State != nil {
		enabled = *vnet.State == clusters.VnetStateEnabled
	}
	
	output := map[string]interface{}{
		"subnet_id":                    vnet.SubnetId,
		"engine_public_ip_id":          vnet.EnginePublicIPId,
		"data_management_public_ip_id": vnet.DataManagementPublicIPId,
		"enabled":                      enabled,
	}
```

Reply:
Good suggestion, I will update this code block